pubsub.go

```diff
@@ -305,13 +319,15 @@ func (p *PubSub) processLoop(ctx context.Context) {
 				log.Warning("closing stream for blacklisted peer: ", pid)
 				close(ch)
 				s.Reset()
+				p.disc.NotifyNewPeer(pid)
 				continue
 			}
 
 			p.rt.AddPeer(pid, s.Protocol())
 
 		case pid := <-p.newPeerError:
 			delete(p.peers, pid)
+			p.disc.NotifyNewPeer(pid)
```

Comment:
A little confused about notifying for a new peer when we got an error. The method is probably wrongly named.

Reply:
See above. Feel free to come up with a new name. We're notifying when a new peer connection attempt has completed, whether with success or failure.